lib/src/main/java/com/ibm/bluemix/appid/android/internal/tokenmanager/TokenManager.java

```diff
@@ -109,21 +130,17 @@ private void retrieveTokens(HashMap<String, String> formParams, final TokenRespo
 			@Override
 			public void onFailure (Response response, Throwable t, JSONObject extendedInfo) {
 				logger.error("Failed to retrieve tokens from authorization server", t);
-
+				String errorDescription = "";
 				try {
 					if (response.getStatus() == 400) {
-						JSONObject responseJSON = new JSONObject(response.getResponseText());
-						if (INVALID_GRANT.equals(responseJSON.getString(ERROR))) {
-							listener.onAuthorizationFailure(new AuthorizationException(responseJSON.getString(ERROR_DESCRIPTION)));
-							return;
-						}
-					} else if (response.getStatus() == 403) {
```

Comment:
this code was added by @sandmman to returns pending user error
https://github.com/ibm-cloud-security/appid-clientsdk-android/commit/d6c9022cae37e9cfb141571f24bbc75bb71b1dda
are we sure we need to remove it?

Reply:
@ishangulhane This should not be removed. We need to propagate the 403 error.